pallets/external-validators-rewards/src/lib.rs

```diff
@@ -245,11 +255,20 @@ pub mod pallet {
     impl<T: Config> tp_traits::OnEraEnd for Pallet<T> {
         fn on_era_end(era_index: EraIndex) {
             if let Some(utils) = Self::generate_era_rewards_utils(era_index, None) {
+                let tokens_inflated = T::EraInflationProvider::get();
+
+                let ethereum_sovereign_account = T::RewardsEthereumSovereignAccount::get();
+                if let Err(err) =
+                    T::Currency::mint_into(&ethereum_sovereign_account, tokens_inflated.into())
+                {
+                    log::error!(target: "ext_validators_rewards", "Failed to mint inflation into Ethereum Soverein Account: {err:?}");
```

Comment:
In case of error (which will never happen) maybe set `tokens_inflated` in the message to 0?

Reply:
yes

Reply:
Changed. However couldn't this be a source of bugs on the Ethereum side if the amount is zero? There must be checks in case the reward amount is zero to not distribute rewards (which would perform a division by zero)

Reply:
Yes but the alternative is rewarding with tokens that don't exist. Let's make sure we have a test for zero amount in ethereum side.

Or don't send a rewards message at all, but not sure if that's better.

Reply:
I feel like not sending a message would be cleaner, but I don't know if the Ethereum side depends on receiving a message every session.

Reply:
It shouldn't break in that case, as there are other error paths in this PR that already don't send a message. So you can change this to not send a message.

Reply:
Other path being the message not being validated? Makes sense